source/BannerlordCoopInstaller/Installer.cs

```diff
@@ -0,0 +1,264 @@
+        /// Check the BannerlordPath and ensure and executable exists. If not, disable the install button
+        /// </summary>
+        private void checkBannerLordPath()
+        {
+            if(File.Exists(this.BannerlordPath.Text + "\\bin\\Win64_Shipping_Client\\Bannerlord.exe"))
+            {
+                this.Install.Enabled = true;
+                this.Status.Text = "Ready to install";
+            }
+            else
+            {
+                this.Status.Text = "Select a Mount && Blade II: Bannerlord install path to continue";
+                this.Install.Enabled = false;
+            }
+        }
+
+        /// <summary>
+        /// Check if MB is installed (through the registry) - Only works in x64.
+        /// </summary>
+        /// <returns>Full path where MB is installed</returns>
+        private string GetBannerLordInstalledPath()
+        {
+
+            string registry_key = @"SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall";
+            using (Microsoft.Win32.RegistryKey key = Registry.LocalMachine.OpenSubKey(registry_key))
+            {
+                foreach (string subkey_name in key.GetSubKeyNames())
+                {
+                    using (RegistryKey subkey = key.OpenSubKey(subkey_name))
+                    {
+                        string name = (string)subkey.GetValue("DisplayName");
+                        if (name != null && name.Equals("Mount & Blade II: Bannerlord"))
+                        {
+                            return subkey.GetValue("installlocation").ToString();
+                        }
+                    }
+                }
+            }
+            return null;
+        }
+
+        /// <summary>
+        /// Initialize the form; create a map of the objects stored as resources in the executable; include the path where it should be installed and extension.
+        /// </summary>
+        public BannerlordCoopInstallerForm()
+        {
+            InitializeComponent();
+            this.BannerlordPath.Text = GetBannerLordInstalledPath();
+            checkBannerLordPath();
+            this.resourceDict = new Dictionary<string, ResourceFile> {
+                {"SubModule", new ResourceFile("xml", @"\") },
+                { "CoopConnectionUIMovie", new ResourceFile( "xml", @"\GUI\Prefabs")},
+                { "Launcher", new ResourceFile("exe", @"\bin\Win64_Shipping_Client") },
+                { "bannerlord_icon", new ResourceFile("ico", @"\") }
+            };
+
+
+        }        
+        
+
+
+        /// <summary>
+        /// Allow user to click the browse button, open a prompt to select a different path.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void PathBrowse_Click(object sender, EventArgs e)
+        {
+            FolderBrowserDialog folderBrowserDialog1 = new FolderBrowserDialog();
+            if (folderBrowserDialog1.ShowDialog() == DialogResult.OK)
```

Comment:
Does the cancel button work? Is that inherit functionality from forms?

Reply:
The functionality of the dialog box is inherited from the second parameter: MessageBoxButtons.YesNo in the declaration.